pulp_ansible/app/downloaders.py

```diff
@@ -90,11 +94,46 @@ async def _run(self, extra_data=None):
         if not self.ansible_auth_url:
             # Galaxy Token
             headers = {"Authorization": self.ansible_token}
+            return await self._run_with_additional_headers(headers)
         else:
-            token = await self.update_token_from_auth_url()
+            return await self._run_with_token_refresh_and_401_retry()
+
+    async def _run_with_token_refresh_and_401_retry(self):
+        """
+        Fetch the response and refresh the Keycloak token if needed when doing so.
+
+        If the fetching of data returns a 401 exception, invalidate the token and try to refresh it
+        once again.
+
+        Returns:
+            DownloadResult: Contains information about the result. See the DownloadResult docs for
+                 more information.
+        """
+        while True:
+            token = await self.get_or_update_token()
             # Keycloak Token
             headers = {"Authorization": "Bearer {token}".format(token=token)}
+            try:
+                return await self._run_with_additional_headers(headers)
+            except ClientResponseError as exc:
+                if exc.status is 401:
```

Comment:
```suggestion
                if exc.status == 401:
```

Reply:
fixing in next push along w/ CI. Thanks!